yt_dlp/extractor/viu.py

```diff
@@ -312,35 +316,36 @@ def _real_extract(self, url):
             'ccs_product_id': video_data['ccs_product_id'],
             'language_flag_id': self._LANGUAGE_FLAG.get(lang_code.lower()) or '3',
         }
-        headers = {
-            'Referer': url,
-            'Origin': url,
-        }
-        try:
+
+        def download_playback():
             stream_data = self._download_json(
-                'https://d1k2us671qcoau.cloudfront.net/distribute_web_%s.php' % country_code,
-                video_id, 'Downloading stream info', query=query, headers=headers)
-            stream_data = self._detect_error(stream_data)['stream']
+                'https://api-gateway-global.viu.com/api/playback/distribute',
+                video_id=video_id,
+                note='Downloading stream info',
+                query=query,
+                headers={
+                    'Authorization': f'Bearer {self.auth_codes[country_code]}',
+                    'Referer': url,
+                    'Origin': url
+                })
+            return self._detect_error(stream_data).get('stream')
+
+        try:
+            stream_data = download_playback()
         except (ExtractorError, KeyError):
             stream_data = None
             if video_data.get('user_level', 0) > 0:
-                user = self._login(country_code, video_id)
+                user = self._login(self._country_code, video_id)
```

Comment:
mistake?
```suggestion
                user = self._login(country_code, video_id)
```

Reply:
Yup I forgot to revert it. was going to change it if there's more to fix.